internal/app/machined/pkg/controllers/k8s/templates.go

```diff
@@ -327,16 +365,14 @@ spec:
     spec:
       affinity:
         podAntiAffinity:
-          preferredDuringSchedulingIgnoredDuringExecution:
-          - weight: 100
-            podAffinityTerm:
-              labelSelector:
-                matchExpressions:
-                - key: k8s-app
-                  operator: In
-                  values:
-                  - kube-dns
-              topologyKey: kubernetes.io/hostname
+          requiredDuringSchedulingIgnoredDuringExecution:
```

Comment:
This will make deploying to a single node impossible. We will need the ability to set the number of replicas.

Reply:
Yes, deploy I not finish. But it is not affect anything.
```shell
NAME      READY   UP-TO-DATE   AVAILABLE   AGE
coredns   1/2     2            1           2m10s

NAME                            READY   STATUS    RESTARTS   AGE
coredns-6bd64fcdc6-8f5ht        0/1     Pending   0          2m55s
coredns-6bd64fcdc6-95rlv        1/1     Running   0          2m55s
```

Reply:
I've add the version tag, to make possible rolling update works.
As I know talos does't upgrade the deployments after **edit machineconfig** command. This feature may be solve rolling update in future.

Reply:
I think our health checks check to see if coredns has been rolled out fully. @smira correct me if I am wrong.

Reply:
Add cluster.coreDNS.replicas = 2 (by default)

Reply:
I feel we should support custom deployments for coreDNS, but I don't know whether this is the right thing to do here, as it kind of forces to set number of replicas for single-node clusters